### ISSUE TITLE
Prevent XMLTool from fetching embedded entities

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#21415]
dotCMS has an XXE vulnerability in the XMLTool. To exploit this flaw, a attacker needs the permission to edit and preview content  and this flaw can be abused to read internal services and possibly files

This section of the documentation explain how to use the XMLTool in the template engine, this allow us to direct write xml content to the application

We need to set options in dom4j not to resolve external entities before we call the `XMLTool.read` method

**Comment by nollymar** [comment]
**Internal QA:** Passed.

I tested accessing a public URL (https://www.w3schools.com/XML/cd_catalog.xml") and the xml content was rendered successfully.

<img width="1393" alt="Screen Shot 2022-01-07 at 12 43 51 PM" src="https://user-images.githubusercontent.com/8741395/148584799-9957aaff-e4a8-4c40-8fff-fe7e3babae08.png">


In addition to that, I ran a similar test but using a private IP. I got this log message:
```
dotcms_1         | 17:38:03.102  WARN  viewtools.XmlTool - read XMLTool Cannot access 
private subnets.  Set ALLOW_ACCESS_TO_PRIVATE_SUBNETS=true to allow @ 
url:GET//demo.dotcms.com/activities/hiking | lang:1 | ip:172.31.0.1 | Admin:false | 
start:01-07-2022 05:33:49 UTC  ref:http://localhost:8080/destinations/ 
>/LIVE/FIELD_CONSTANT/10ca3189-8d72-4d63-9b0b-9f15fe802482.fields
```
and the xml wasn't rendered.

<img width="1243" alt="Screen Shot 2022-01-07 at 12 42 58 PM" src="https://user-images.githubusercontent.com/8741395/148584818-f7156d9c-5f01-412d-83ee-17ae241cadd4.png">

**Comment by bryanboza** [comment]
Fixed, tested on release 21.12 // Docker // FF 

Tested with the documentation and this is working as expected
